dali/dfu/dfurun.cpp

```diff
@@ -1700,12 +1710,35 @@ class CDFUengine: public CInterface, implements IDFUengine
                             }
                         }
                         else {
-                            fsys.copy(srcFile,dstFile,recovery, recoveryconn, filter, opttree, &feedback, &abortnotify, dfuwuid);
-                            if (!abortnotify.abortRequested()) {
-                                if (needrep)
-                                    replicating = true;
+                            bool performCopy = true;
+                            if (options->getEnsure())
+                            {
+                                if (ensureLfnAlreadyPublished)
+                                    performCopy = false;
```

Comment:
@jakesmith Just to clarify, this will not do a copy if the meta data is there, but a DFU operation adds the meta data last, so the parts should be there if the file was created by DFU?
I guess then the one scenario that might get confusing is if they first tried to get roxie to copy the file which would create the meta data, but roxie didn't and they tried to do an ensure.
I guess I'm wondering if we should still check dstFile->existsPhysicalPartFiles(0)?

Reply:
Not sure.
If Roxie copied the meta data only, then yes, this 'ensure' would think the file was already there and skip the copy.


>but roxie didn't and they tried to do an ensure

In that scenario, why didn't roxie copy physicals, and/or how do yo know if it's in progress and is currently doing so?
Even if it checked physicals, would they be there (with correct names) if Roxie was copying them, until it had finished copying?
@afishbeck

Reply:
@afishbeck - refreshing reviewing status, per conversation above.

Reply:
I'm thinking in hypotheticals and I'm not sure that scenario would happen in the real world.  I'll finish review as is, we could always cover that case later if it did come up.